web/src/WifiSelector.jsx

```diff
@@ -0,0 +1,276 @@
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import React, { useEffect, useState } from "react";
+import {
+  Button,
+  Card,
+  CardBody,
+  Radio,
+  Spinner,
+  Split,
+  SplitItem,
+  Text
+} from "@patternfly/react-core";
+
+import {
+  EOS_LOCK as LockIcon,
+  EOS_SIGNAL_CELLULAR_ALT as SignalIcon
+} from "eos-icons-react";
+
+import Popup from "./Popup";
+import Center from "./Center";
+import WifiNetworkMenu from "./WifiNetworkMenu";
+import WifiConnectionForm from "./WifiConnectionForm";
+import { useInstallerClient } from "./context/installer";
+import { ConnectionState } from "./client/network/model";
+import { NetworkEventTypes } from "./client/network";
+
+const networksFromValues = (networks) => Object.values(networks).flat();
+const baseHiddenNetwork = { ssid: undefined, hidden: true };
+
+const networkState = (state) => {
+  switch (state) {
+    case ConnectionState.ACTIVATING:
+      return 'Connecting';
+    case ConnectionState.ACTIVATED:
+      return 'Connected';
+    case ConnectionState.DEACTIVATING:
+      return 'Disconnecting';
+    case ConnectionState.DEACTIVATED:
+      return 'Disconnected';
+    default:
+      return "";
+  }
+};
+
+function WifiSelector({ isOpen = false, onClose }) {
+  const client = useInstallerClient();
+  const [networks, setNetworks] = useState([]);
+  const [connections, setConnections] = useState([]);
+  const [activeConnections, setActiveConnections] = useState(client.network.activeConnections());
+  const [selectedNetwork, setSelectedNetwork] = useState(null);
+  const [activeNetwork, setActiveNetwork] = useState(null);
+  const [showHiddenForm, setShowHiddenForm] = useState(false);
+
+  const switchSelectedNetwork = (network) => {
+    setShowHiddenForm(network === baseHiddenNetwork);
+    setSelectedNetwork(network);
+  };
+
+  useEffect(() => {
+    client.network.connections().then(setConnections);
+  }, [client.network]);
+
+  useEffect(() => {
+    const loadNetworks = async () => {
+      const knownSsids = [];
+
+      return client.network.accessPoints()
+        .sort((a, b) => b.strength - a.strength)
+        .reduce((networks, ap) => {
+          // Do not include networks without SSID
+          if (!ap.ssid || ap.ssid === "") return networks;
+          // Do not include "duplicates"
+          if (knownSsids.includes(ap.ssid)) return networks;
```

Comment:
What about filtering networks you cannot configure? For instance, we cannot set up a WEP network, right? Alternatively, we could show them, but we should not allow trying to connect.